### PR TITLE
[docs] Add pie chart demo with a center label

### PR DESCRIPTION
**Pull request by giladappsforce**
- [ ✅ ] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

Added a pieCenter slot and docs to show how to use it

Screenshot 2023-09-04 at 15 04 12

### CODE REVIEW

File: docs/data/charts/pie/pie.md [hunk 1]
```diff
@@ -68,6 +68,14 @@ Arcs with angles smaller than the value (in deg) will not have labels.
 
 {{"demo": "PieArcLabel.js"}}
 
+## Center label
+
+You can add a label to the center of a pie chart.
+To do so, there's a slot called `arcCenter`.
```
Comment (giladappsforce) [inline docs/data/charts/pie/pie.md:74]:
```suggestion
To do so, there's a slot called `pieCenter`.
```

**Comment by giladappsforce** [comment]
https://github.com/mui/mui-x/issues/10222

**Review by LukasTy — COMMENTED** [review]
Thank you for your contribution! 🙏 

I've looked into the suggested change and think that going with a composition example for such cases looks like a better approach.

What do you think, would it cover your use cases? 🤔 

I've prepared the following suggestion for a demo.
Feel free to update the PR accordingly if you agree with my suggestion. 😉 

```diff
diff --git a/docs/data/charts/pie-demo/pie-demo.md b/docs/data/charts/pie-demo/pie-demo.md
index e2b94d314..009bba937 100644
--- a/docs/data/charts/pie-demo/pie-demo.md
+++ b/docs/data/charts/pie-demo/pie-demo.md
@@ -25,3 +25,7 @@ title: Charts - Pie demonstration
 ## PieChartWithPaddingAngle
 
 {{"demo": "PieChartWithPaddingAngle.js"}}
+
+## PieChartWithCenterLabel
+
+{{"demo": "PieChartWithCenterLabel.js"}}

```

And add a new demo file with these contents:
```tsx
import * as React from 'react';
import { PieChart } from '@mui/x-charts/PieChart';
import { useDrawingArea } from '@mui/x-charts/hooks/useDrawingArea';
import { styled } from '@mui/material/styles';

const data = [
  { value: 5, label: 'A' },
  { value: 10, label: 'B' },
  { value: 15, label: 'C' },
  { value: 20, label: 'D' },
];

const size = {
  width: 400,
  height: 200,
};

const StyledText = styled('text')(({ theme }) => ({
  fill: theme.palette.text.primary,
  textAnchor: 'middle',
  dominantBaseline: 'central',
  fontSize: 20,
}));

function PieCenterLabel({ children }: { children: React.ReactNode }) {
  const { width, height, left, top } = useDrawingArea();
  return (
    <StyledText x={left + width / 2} y={top + height / 2}>
      {children}
    </StyledText>
  );
}

export default function PieChartWithCenterLabel() {
  return (
    <PieChart series={[{ data, innerRadius: 80 }]} {...size}>
      <PieCenterLabel>Center label</PieCenterLabel>
    </PieChart>
  );
}
```

**Comment by LukasTy** [comment]
> I moved the doc part to the demo that does make more sense.

My main idea was that I'm not sure we should be introducing a new slot for such a thing that could easily be achieved with even fewer lines of compositional code.
In essence, I was suggesting to only provide a demo on how to achieve your desired result using composition. 🤔

**Comment by giladappsforce** [comment]
> My main idea was that I'm not sure we should be introducing a new slot for such a thing that could easily be achieved with even fewer lines of compositional code.
> 
> In essence, I was suggesting to only provide a demo on how to achieve your desired result using composition. 🤔 

Yeah after wrapping my head around everything you said I strongly agree 😅

I'll adjust the PR

**Comment by giladappsforce** [comment]
> My main idea was that I'm not sure we should be introducing a new slot for such a thing that could easily be achieved with even fewer lines of compositional code. In essence, I was suggesting to only provide a demo on how to achieve your desired result using composition. 🤔

Fixed it 🙃

**Review by LukasTy — APPROVED** [review]
Thank you for the contribution and landing on an agreement for the solution! 👍 🎉 🙏

**Pull request merged**